webrtc/videoroom/assets/src/room.ts

```diff
@@ -79,7 +79,21 @@ export class Room {
     );
   }
 
-  public init = async () => {
+  public join = () => {
+    this.init()
```

Comment:
Can't you make the join function async and then use `await` and `try/catch` syntax instead of `then` and `catch`? Promises in JS are eager instead of being lazy so it will eventually execute, you just have to use await if you want it to finish executing before certain point in code. So calling `room.join` will behave the same being and not being asynchronous.

Reply:
Sure, makes sense, I will change this